src/ol/layer/STAC.js

```diff
@@ -710,6 +708,10 @@ class STACLayer extends LayerGroup {
       );
     }
 
+    if (this.buildTileUrlTemplate_ && !this.useTileLayerAsFallback_) {
+      return await this.addTileLayerForImagery_(asset);
+    }
+
```

Comment:
Why was this code block moved?

Reply:
... or might just be a race condition. tbc

Reply:
@m-mohr at least are necessary these steps (or info in options) before asking for tile
```
        const sourceInfo = getGeoTiffSourceInfoFromAsset(asset, this.bands_);
        /**
         * @type {import("ol/source/GeoTIFF.js").Options}
         */
        let options = {
            sources: [sourceInfo],
        };
        const projection = await getProjection(asset);
        if (projection) {
            options.projection = projection;
        }
```
projection are enough to generate a good tiling schema, but without having sopurceinfo no rendering is done

Reply:
found the issue! without modifying the code, the  `buildTileUrlTemplate` have to be a promise (or should be in my implementation) in stac-browser/config.js
e.g.    
`buildTileUrlTemplate: async (asset) => {...............},`

Reply:
then reverted this commit